developers/blobstream-rollups.md

```diff
@@ -0,0 +1,399 @@
+---
+description: Learn how to build rollups that use Blobstream
+---
+
+## Introduction
+
+[Blobstream](https://blog.celestia.org/introducing-blobstream/)
+is the first data availability solution for EVM chains that securely
+scales with the number of users.
+It allows rollups to post their data on Celestia while proving their availability in the rollup settlement contract.
+
+This document will outline a few ways to build optimistic or zk-rollups that post their data to Celestia
+and use Blobstream to prove that data's availability.
+
+## Concepts
+
+### Share commitment
+
+The [share commitment](https://celestiaorg.github.io/celestia-app/specs/data_square_layout.html#blob-share-commitment-rules) is a commitment over the data
+contained in the [PayForBlob](https://github.com/celestiaorg/celestia-app/blob/v1.0.0-rc2/proto/celestia/blob/v1/tx.proto#L16-L31) transaction.
+This commitment allows
+proving that the corresponding data exists on Celestia [efficiently](https://github.com/celestiaorg/celestia-app/blob/main/docs/architecture/adr-011-optimistic-blob-size-independent-inclusion-proofs-and-pfb-fraud-proofs.md).
+
+#### Proof details
+
+To prove that the data corresponding to a share commitment was posted to Celestia using Blobstream,
+the following proofs need to be verified:
+
+- share inclusion proof to the share commitment: meaning creating two merkle proofs:
+  - share merkle proof up to the [subtree root](https://celestiaorg.github.io/celestia-app/specs/data_square_layout.html#blob-share-commitment-rules)
+    corresponding to that share
+  - subtree root merkle proof to the [share commitment](https://celestiaorg.github.io/celestia-app/specs/data_square_layout.html#blob-share-commitment-rules)
+- share commitment inclusion proof to the data root tuple root: meaning four merkle proofs:
```

Comment:
when would users of blobstream create one of these again? do we know if this is less efficient than just using a share proof to the data commitment

Reply:
yes, it's less efficient because you need an extra proof of the share to the share commitment